### PR TITLE
Reduce includes

### PR DESCRIPTION
**Pull request by jkflying**
Reduce the number of un-needed includes and chained dependencies, which should help to keep compile times reasonable as the codebase grows.

I also pulled the output from the local planner into a separate file so that not everything depends on the local planner when it doesn't need to.

### CODE REVIEW

File: local_planner/src/nodes/local_planner_node.cpp [hunk 1]
```diff
@@ -745,7 +761,7 @@ void LocalPlannerNode::publishGround() {
 }
 
 void LocalPlannerNode::printPointInfo(double x, double y, double z) {
-  geometry_msgs::PoseStamped drone_pos = local_planner_.getPosition();
+  geometry_msgs::PoseStamped drone_pos = local_planner_->getPosition();
   int beta_z = floor(
```
Comment (nicovanduijn) [inline local_planner/src/nodes/local_planner_node.cpp]: isn't there a function in common.cpp for this?

File: local_planner/src/nodes/planner_functions.cpp [hunk 2]
```diff
@@ -29,20 +33,20 @@ double adaptSafetyMarginHistogram(double dist_to_closest_point,
...
     double min_cloud_size, double min_dist_backoff, Box histogram_box,
-    const Eigen::Vector3f &position, double min_realsense_dist) {
+    const Eigen::Vector3f& position, double min_realsense_dist) {
   cropped_cloud.points.clear();
   cropped_cloud.width = 0;
   distance_to_closest_point = HUGE_VAL;
   float distance;
   counter_backoff = 0;
 
   for (size_t i = 0; i < complete_cloud.size(); ++i) {
```
Comment (nicovanduijn) [inline local_planner/src/nodes/planner_functions.cpp]: outer loop could be range-based as well

File: local_planner/src/nodes/planner_functions.cpp [hunk 3]
```diff
@@ -117,11 +121,12 @@ void calculateFOV(double h_fov, double v_fov, std::vector<int> &z_FOV_idx,
...
-                        pcl::PointCloud<pcl::PointXYZ> reprojected_points,
-                        std::vector<double> reprojected_points_age,
-                        std::vector<double> reprojected_points_dist,
-                        geometry_msgs::PoseStamped position) {
+void propagateHistogram(
+    Histogram& polar_histogram_est,
+    const pcl::PointCloud<pcl::PointXYZ>& reprojected_points,
+    const std::vector<double>& reprojected_points_age,
+    const std::vector<double>& reprojected_points_dist,
+    geometry_msgs::PoseStamped position) {
```
Comment (nicovanduijn) [inline local_planner/src/nodes/planner_functions.cpp]: could this be passed by reference? It's quite big actually, 7 doubles plus the header

File: local_planner/src/nodes/local_planner_node.cpp [hunk 4]
```diff
@@ -488,13 +502,15 @@ void LocalPlannerNode::publishBox() {
   box.pose.position.x = drone_pos.pose.position.x;
   box.pose.position.y = drone_pos.pose.position.y;
   box.pose.position.z = drone_pos.pose.position.z;
+                        0.5 * (local_planner_->histogram_box_.zsize_up_ -
```
[the remark this answers is not in the document]
Comment (jkflying) [reply local_planner/src/nodes/local_planner_node.cpp]: Rebase issues... I thought I caught this.

File: local_planner/src/nodes/planner_functions.cpp (hunk 3 above)
Comment (jkflying) [reply local_planner/src/nodes/planner_functions.cpp]: done

File: local_planner/src/nodes/planner_functions.cpp (hunk 2 above)
Comment (jkflying) [reply local_planner/src/nodes/planner_functions.cpp]: done

File: local_planner/src/nodes/local_planner_node.cpp (hunk 1 above)
Comment (jkflying) [reply local_planner/src/nodes/local_planner_node.cpp]: I'd rather not change that in this PR

**Pull request merged**